gflownet/policy/multihead_tree.py

```diff
@@ -0,0 +1,502 @@
+from typing import Optional
+
+import torch
+import torch_geometric
+from torch_geometric.data import Batch
+from torch_geometric.nn import global_add_pool
+from torch_geometric.utils import unbatch
+
+from gflownet.envs.tree import Attribute, Stage, Tree
+from gflownet.policy.base import Policy
+
+
+class Backbone(torch.nn.Module):
```

Comment:
Could the Backbone of the simple tree be reused here? Alternatively, could we perhaps have a `tree.py` with the common code for both? Not sure what would be better.

Reply:
So right now the backbones are actually slightly different - one uses `GINEConv` layers, and the other `GCNConv` (and there are some differences in how they have to be constructed, `GINEConv` requires specifying an MLP model used by the layer, and also uses edge features.

I will simplify that in the end (and keep a single backbone class), but I haven't figured out yet which is better.

Reply:
Ok fine. We can keep them separate for now, no problem. And possibly combine things later.